crates/biome_js_analyze/src/lint/nursery/use_date_now.rs

```diff
@@ -0,0 +1,276 @@
+    /// ```
+    /// ```js,expect_diagnostic
+    /// const foo = new Date() * 2;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const foo = Date.now();
+    /// ```
+    /// ```js
+    /// const foo = Date.now() * 2;
+    /// ```
+    ///
+    pub UseDateNow {
+        version: "next",
+        name: "useDateNow",
+        language: "js",
+        sources: &[RuleSource::EslintUnicorn("prefer-date-now")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseDateNow {
+    type Query = Ast<JsNewOrCallExpression>;
+    type State = (AnyJsExpression, UseDateNowIssueKind);
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let expr = ctx.query();
+
+        match expr {
+            JsNewOrCallExpression::JsCallExpression(call_expr) => get_date_method_issue(call_expr),
+            JsNewOrCallExpression::JsNewExpression(expr) => get_new_date_issue(expr),
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, (node, kind): &Self::State) -> Option<RuleDiagnostic> {
+        let message = match kind {
+            UseDateNowIssueKind::ReplaceMethod(method) => format!("new Date().{}", method),
+            UseDateNowIssueKind::ReplaceConstructor => "new Date()".to_string(),
+            UseDateNowIssueKind::ReplaceNumberConstructor => "Number(new Date())".to_string(),
+        };
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Use "<Emphasis>"Date.now()"</Emphasis>" instead of "<Emphasis>{message}</Emphasis>"."
+            },
+        ).note(
+            markup! {
+                <Emphasis>"Date.now()"</Emphasis>" is more readable and also avoids unnecessary instantiation of "<Emphasis>"Date"</Emphasis>"object."
+            }
+            .to_owned(),
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, (node, _): &Self::State) -> Option<JsRuleAction> {
+        let mut mutation = ctx.root().begin();
+
+        let args = make::js_call_arguments(
+            make::token(T!['(']),
+            make::js_call_argument_list([], []),
+            make::token(T![')']),
+        );
+
+        let mut date_now_expr = make::js_static_member_expression(
+            make::js_identifier_expression(make::js_reference_identifier(make::ident("Date")))
+                .into(),
+            make::token(T![.]),
+            make::js_name(make::ident("now")).into(),
+        );
+
+        if does_node_needs_space_before_child(&node.syntax().parent()?) {
+            // Make fake token to get leading trivia
+            let leading_trivia = make::token_decorated_with_space(T![=])
+                .leading_trivia()
+                .pieces();
+
+            // In case `await +a` this produces double  space `await  +a`.
+            // TODO. Find a way to avoid this.
+            date_now_expr = date_now_expr
```

Comment:
I wish there was more ergnomic way to fix this. I imagine a lot of rules are/will be impacted by this. Unicorn provides [fixSpaceAroundKeyword](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/810575620e3d9d90c60c2d6c5f2098c0a7b0348d/rules/fix/fix-space-around-keywords.js#L12-L34) but not sure how would that fit into biome.

Reply:
The formatter, that's for sure

Reply:
You mean we would call formatter from lint rules? Conceptually it makes sense. Is there any precedent currently in the codebase?

Reply:
I might have misunderstood the comment, I'm sorry about that. You can ignore my comment